application/models/Question.php

```diff
@@ -168,7 +168,7 @@ public function rules()
             return 'N';
         });
         /* Don't save empty or 'core' question theme name */
-        $aRules[] = ['question_theme_name', 'questionThemeNameValidator'];
+        $aRules[] = ['question_theme_name', 'filter', 'filter' =>  [$this, 'questionThemeNameValidator'] ];
```

Comment:
Is this rule applied only when question_them_name is modified btw? Is Yii smart enough to optimize that?

Reply:
No,

Another way to broke

1. Import a questiontheme
2. Create a question with this theme
3. Delete the theme

question theme was not updated but broke.

Reply:
> No,

:(

Reply:
> Ah but that's missing logic in question theme delete process, then.

Sorry : i mean : Delete the Theme directory.

Else: before it was in afterFind, moved to when save : action is done ONLY by admin. 

Why create an optimization here ?

Reply:
> > Ah but that's missing logic in question theme delete process, then.
> 
> Sorry : i mean : Delete the Theme directory.
> 
> Else: before it was in afterFind, moved to when save : action is done ONLY by admin.
> 
> Why create an optimization here ?

Just avoiding needless queries to the db.

Reply:
> > > Ah but that's missing logic in question theme delete process, then.
> > 
> > 
> > Sorry : i mean : Delete the Theme directory.
> > Else: before it was in afterFind, moved to when save : action is done ONLY by admin.
> > Why create an optimization here ?
> 
> Just avoiding needless queries to the db.

But here : only when save action by admin, it's already a save action.

If we can not fix here : when we fix it ?

Reply:
> Just avoiding needless queries to the db.

Else : it's important when there a 100 queries in same page : https://bugs.limesurvey.org/view.php?id=18474 (and related)

But here : just a sample query …

Maybe we can improve it, but still need it …

Alternative solution : fix it when load page by admin ? In QuestionController ? 
But i think Load QuestionController happen more than save.